### PR TITLE
Solutions to coding problems for my internship application

### PR DESCRIPTION
**Pull request by dizhen**
Please see the detailed description of the web app and my answers to the additional questions in the README.txt file.

Thanks,
Zhen

### CODE REVIEW

**Comment by mrcaron** [comment]
Thanks, @dizhen! I'm a bit behind due to an overwhelming amount of work this week, but I'll be in touch next week as I review solutions and resumes.

**Comment by mrcaron** [comment]
Hi @dizhen, I've been out sick and am now just getting back into things. Are you still available for interviews?

**Comment by dizhen** [comment]
Hi Mike,

Yes, I am available for interviews. I'll be happy to chat about my code with you.

Thanks,
Zhen

On Mar 2, 2018, at 12:23, Michael Caron <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi @dizhen<https://github.com/dizhen>, I've been out sick and am now just getting back into things. Are you still available for interviews?

**Comment by mrcaron** [comment]
Hi Zhen, can you indicate your preferred interview times on this document? Please choose times which are not already taken. :)

https://docs.google.com/document/d/1F2XUtdDo9_81fU2VzonBMpHtZvOnFCzX8oU0OGB9NW8/edit?usp=sharing.

**Comment by dizhen** [comment]
Good evening Mike,

Thank you for getting back to me! It seems like none of the empty time slots works well for me. I'm fully booked tomorrow (3/6) and this Thursday (3/8). I'm available this Wednesday or Friday after 10 a.m. (because I don't have a car, it'll take a while to commute from my home in Verona to the SolidWorks office at east town).

Can I take March 7th 10:30 or 12:30 if William ended up taking one of his preferences? Otherwise, I'm pretty open next week Monday through Wednesday if that works better for you. I sincerely apologize for the inconvenience.

Best,
Zhen


On Mar 5, 2018, at 22:10, Michael Caron <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi Zhen, can you indicate your preferred interview times on this document? Please choose times which are not already taken. :)

https://docs.google.com/document/d/1F2XUtdDo9_81fU2VzonBMpHtZvOnFCzX8oU0OGB9NW8/edit?usp=sharing.

**Comment by mrcaron** [comment]
Yeah, let's do next Monday morning? Does 10:00 - 11:30 work for you?

**Comment by dizhen** [comment]
That works. Thank you for being so flexible. See you then!

Best,
Zhen

On Mar 7, 2018, at 09:07, Michael Caron <notifications@github.com<mailto:notifications@github.com>> wrote:


Yeah, let's do next Monday morning? Does 10:00 - 11:30 work for you?

File: index.js [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+        $input.prop('disabled', true);
+        _disableAllButtons();
+        _hideAllAlerts();
+
+        const $output = $('#output-palindrome-checker');
+        const intervalID = setInterval(checkPalindromeAnimation, DEFAULT_PALINDROME_ANIMATION_INTERVAL);
+        var input = $input.val(), leftHalf = '', rightHalf = '', isPalindrome = true;
+
+        function checkPalindromeAnimation() {
```
Comment (mrcaron) [inline index.js:76]: I loved this. Such a neat way to visualize what's going on.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,197 @@
+/**
```
Comment (mrcaron) [inline index.js:1]: Really nice, well-constructed javascript file.

File: index.js [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+ */
+
+"use strict";
+
+const TEXTAREA_MAX_HEIGHT = 400;
+const DEFAULT_TIMEOUT = 3000;
+const DEFAULT_REVERSE_ANIMATION_INTERVAL = 100;
+const DEFAULT_PALINDROME_ANIMATION_INTERVAL = 200;
+
+var stringManipulator = (function ($) {
```
Comment (mrcaron) [inline index.js:16]: Classic javascript object-orientation; easy to follow and update with changes.

File: README.txt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+For a complete user experience, I decided to build a web app that allows the users to input strings/JSON
+texts and visualizes the result of the following three problems:
+- reverse a string
+- check if a string is palindrome
+- print out a JSON object
+
+At first, I wrote a Java program that interacts with users through command line. After considering
+some potential use cases of the program, I think it fits the web technology better. I used jQuery
+and Bootstrap 4 in the project. I haven't been doing web development for a while, so I did lots of
+googling and research to refresh my memory while writing the app. I enjoyed the learning process.
```
Comment (mrcaron) [inline README.txt:13]: Glad you enjoyed it! I meant this to be a fun exercise :)

**Pull request closed without merging**